tests/test_hook.rs

```diff
@@ -55,8 +56,43 @@ fn test_no_original_hook() {
     initialize_hook(Configuration {
         force_capture: true,
         keep_original_hook: false,
+        ..Default::default()
     });
     let _ = panic::catch_unwind(|| panic!("Test"));
 
     assert_eq!(*ran_hook.lock().unwrap(), false);
 }
+
+#[test]
+fn test_flush_logger() {
+    struct Logger {
+        pub flushed: Arc<Mutex<bool>>,
+    }
+
+    impl log::Log for Logger {
+        fn enabled(&self, _metadata: &log::Metadata) -> bool {
+            unimplemented!()
+        }
+
+        fn log(&self, _record: &log::Record) {
+            unimplemented!()
+        }
+
+        fn flush(&self) {
+            *self.flushed.lock().unwrap() = true;
+        }
+    }
+
+    static LOGGER: LazyLock<Logger> = LazyLock::new(|| Logger {
+        flushed: Arc::new(Mutex::new(false)),
+    });
+
+    initialize_hook(Configuration {
+        force_capture: true,
+        keep_original_hook: true,
+        logger: Some(&*LOGGER),
+    });
+    let _ = panic::catch_unwind(|| panic!("Test"));
+
+    assert_eq!(*LOGGER.flushed.lock().unwrap(), true);
```

Comment:
```suggestion
    assert!(*LOGGER.flushed.lock().unwrap());
```

(Need to fix that in the others as well, it was all a bit rushed)

Reply:
Updated!